### PR TITLE
feat(app, app-shell, app-shell-odd): remove analytics modals and migrate analytics to default to on

### PR DESCRIPTION
**Pull request by shlokamin**
# Overview
This PR migrates the desktop app and ODD's analytics config to default to on. It also removes the `seenOptIn` config value since it is no longer used. Finally, it removes the app analytics modals that show up when you open the desktop app/ODD for the first time.

closes AUTH-529, AUTH-530, AUTH-532, AUTH-533

# Test Plan
This one is sort of hard to test — make sure the code looks good and that the migration does what you'd expect. We'll just have to make sure to battle test this one in QA cc @nusrat813 and @y3rsh 

# Changelog

- Default desktop app mixpanel analytics to on
- Default ODD mixpanel analytics to on
- Remove desktop app's analytics modal

# Review requests
See test plan

# Risk assessment

Medium

### CODE REVIEW

File: app-shell/src/config/__tests__/migrate.test.ts [hunk 1]
```diff
@@ -23,181 +23,189 @@ import {
...
+  it('should migration version 21 to latest', () => {
     const v21Config = MOCK_CONFIG_V21
     const result = migrate(v21Config)
 
     expect(result.version).toBe(NEWEST_VERSION)
-    expect(result).toEqual(v21Config)
+    expect(result).toEqual(MOCK_CONFIG_V22)
+  })
+  it('should keep version 22', () => {
+    const v21Config = MOCK_CONFIG_V22
```
Comment (koji) [inline app-shell/src/config/__tests__/migrate.test.ts:205]:
```suggestion
    const v22Config = MOCK_CONFIG_V22
```

File: app-shell/src/config/__tests__/migrate.test.ts [hunk 2]
```diff
@@ -23,181 +23,189 @@ import {
...
+    const result = migrate(v21Config)
```
Comment (koji) [inline app-shell/src/config/__tests__/migrate.test.ts:206]:
```suggestion
    const result = migrate(v22Config)
```

**Review by koji — APPROVED** [review]
the changes look good to me.

File: app-shell/src/config/__tests__/migrate.test.ts (hunk 2 above)
Comment (shlokamin) [reply app-shell/src/config/__tests__/migrate.test.ts:206]: oops i forgot to push these changes before i merged... i will do this in another PR

**Pull request merged**